init.lua

```diff
@@ -1186,10 +1200,38 @@ function PaperWM:moveWindow(window, frame)
     end)
 end
 
+---add or remove focused window from the floating layer and retile the space
+function PaperWM:toggleFloating()
+    local window = Window.focusedWindow()
+    if not window then
+        self.logger.d("focused window not found")
+        return
+    end
+
+    local id = window:id()
+    local space = nil
+    is_floating[id] = not is_floating[id]
```

Comment:
when removing a floating window, should we remove the window id from the `is_floating` list, instead of setting the dictionary value to false? It looks like this `is_floating` list might grow forever.

Reply:
Right you are, I got lazy ;)  Fixed.